### PR TITLE
refactor: [M3-8648] – Migrate several components to `ui` package prior to migrating `Paper`

### PR DESCRIPTION
**Pull request by dwiley-akamai**
## Description 📝
Migrate the following components from `manager` to `ui`:

- `FormControl`
- `FormHelperText`
- `Input`
- `InputAdornment`
- `InputLabel`

This PR is the result of a cascade of dependencies when I initially tried migrating `Paper`, which imports `FormHelperText`. I wanted to avoid a giant 200+ file PR so this PR sets the groundwork for a follow-up PR focused on migrating `Paper` by handling the cascade of dependencies and cleaning up TS errors.

## Target release date 🗓️
11/12/24

### Verification steps
- Ensure there are no regressions or crashes in Cloud
- All checks should pass (flakes aside)

## As an Author I have considered 🤔
- [X] 👀 Doing a self review
- [X] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [X] 🤏 Splitting feature into small PRs
- [X] ➕ Adding a changeset
- [X] 🧪 Providing/Improving test coverage
- [X] 🔐 Removing all sensitive information from the code and PR description
- [X] 🚩 Using a feature flag to protect the release
- [X] 👣 Providing comprehensive reproduction steps
- [X] 📑 Providing or updating our documentation
- [X] 🕛 Scheduling a pair reviewing session
- [X] 📱 Providing mobile support
- [X] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/TextField.test.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
+import { InputAdornment } from '@linode/ui/';
```
Comment (hkhalil-akamai) [inline packages/manager/src/components/TextField.test.tsx:1]: Source of unit test failure

```suggestion
import { InputAdornment } from '@linode/ui';
```

**Comment by dwiley-akamai** [comment]
> I think I set the wrong precedent with `Tooltip`, but what do we think about making a directory for components with multiple supporting files?
> 
> For example, we could create a directory `FormControl` which contains both `FormControl.tsx` and `FormControl.stories.tsx`, along with an `index.ts` file to re-export the component? This would help in keeping the `components` directory neat.

We have a mix of both approaches in `manager/src/components` right now. We could go either way with it but since it's just the component and its story (I don't think unit test files are getting migrated, right?), I don't mind not having the extra directory layer.

**Comment by hkhalil-akamai** [comment]
> We have a mix of both approaches in `manager/src/components` right now. We could go either way with it but since it's just the component and its story (I don't think unit test files are getting migrated, right?), I don't mind not having the extra directory layer.

Sounds good, I'd be interested to hear what others think though. I believe we are migrating unit tests though, for components that have them.

**Comment by mjac0bs** [comment]
> Sounds good, I'd be interested to hear what others think though. I believe we are migrating unit tests though, for components that have them.

Just chiming in to say that I like directories for components and their related files a lot. The loose files kind of stress me out. 😅 We actually recommend the dirs as a best practice in our docs: https://linode.github.io/manager/development-guide/01-repository-structure.html#where-to-go-for-common-tasks.

Screenshot 2024-10-25 at 11 43 58 AM

**Review by hkhalil-akamai — APPROVED** [review]
Components working as expected. Approved pending decision on whether to make component folders.

**Comment by dwiley-akamai** [comment]
> Just chiming in to say that I like directories for components and their related files a lot. The loose files kind of stress me out. 😅 We actually recommend the dirs as a best practice in our docs: https://linode.github.io/manager/development-guide/01-repository-structure.html#where-to-go-for-common-tasks.

Good catch that it's in the docs, I think we should abide by that then.

**Review by jaalah-akamai — APPROVED** [review]

File: packages/manager/src/components/TextField.tsx [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import { InputLabel } from '@linode/ui';
```
[the remark this answers is not in the document]
Comment (dwiley-akamai) [reply packages/manager/src/components/TextField.tsx:1]: Definitely, thanks for catching -- that was an oversight from the import paths being auto-updated by the editor

**Pull request merged**